exp-android-sdk/src/main/java/com/scala/exp/android/sdk/Runtime.java

```diff
@@ -16,66 +20,51 @@
 public class Runtime extends Exp{
 
     /**
-     * Start with device credentials Host,UUID,Secret TODO make this ExpObservable
-     * @param host
+     * Start with device credentials Host,UUID,Secret
+     * @param consumerApp
      * @param uuid
      * @param secret
-     * @return
+     * @return String token
      */
-    public static Observable<Boolean> start(String host, String uuid, String secret){
+    private static String createToken(String uuid, String secret,boolean consumerApp){
         Map<String,Object> header = new HashMap<String,Object>();
         header.put(Utils.TYP, Utils.JWT);
         Map<String,Object> payload = new HashMap<String,Object>();
-        payload.put(Utils.UUID, uuid);
-        String token = Jwts.builder().setHeader(header).setClaims(payload).signWith(SignatureAlgorithm.HS256, secret.getBytes()).compact();
-        AppSingleton.getInstance().setToken(token);
-        return  ExpService.init(host,token).flatMap(new Func1<Boolean, Observable<Boolean>>() {
-            @Override
-            public Observable call(Boolean aBoolean) {
-                socketManager = new SocketManager();
-                return socketManager.startSocket();
-            }
-        });
-    }
+        if(consumerApp){
```

Comment:
instead of passing in a boolean and checking it, passing in the uuid field name would make this function more reusable later (ie. createToken(myUuid, mySecret, Utils.CONSUMER_APP_UUID); )